fast/stages/02-networking-nva/nva.tf

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  _subnets = var.data_dir == null ? tomap({}) : {
+    for f in fileset("${var.data_dir}/subnets", "**/*.yaml") :
+    trimsuffix(basename(f), ".yaml") => yamldecode(file("${var.data_dir}/subnets/${f}"))
+  }
+  subnets = merge(
+    { for k, v in local._subnets : "${k}-cidr" => v.ip_cidr_range },
+    { for k, v in local._subnets : "${k}-gw" => cidrhost(v.ip_cidr_range, 1) }
+  )
+}
+
+# europe-west1
+
+module "nva-template-ew1" {
+  source         = "../../../modules/compute-vm"
+  project_id     = module.landing-project.project_id
+  name           = "nva-template"
+  zone           = "europe-west1-b"
+  tags           = ["nva"]
+  can_ip_forward = true
+  network_interfaces = [
+    {
+      network    = module.landing-untrusted-vpc.self_link
+      subnetwork = module.landing-untrusted-vpc.subnet_self_links["europe-west1/landing-untrusted-default-ew1"]
+      nat        = false
+      addresses  = null
+    },
+    {
+      network    = module.landing-trusted-vpc.self_link
+      subnetwork = module.landing-trusted-vpc.subnet_self_links["europe-west1/landing-trusted-default-ew1"]
+      nat        = false
+      addresses  = null
+    }
+  ]
+  boot_disk = {
+    image = "projects/debian-cloud/global/images/family/debian-10"
+    type  = "pd-balanced"
+    size  = 10
+  }
+  create_template = true
+  metadata = {
+    startup-script = templatefile("${path.module}/data/nva-startup-script-ew1.tftpl", local.subnets)
```

Comment:
I would pass in explicit values to `templatefile()` to make it obvious what the template depends on.

Reply:
done